core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -1132,7 +1134,18 @@ public String getPluginRepositoryBaseUrl() {
 
         private void testConnection(URL url) throws IOException {
             try {
-                Util.copyStreamAndClose(ProxyConfiguration.open(url).getInputStream(),new NullOutputStream());
+                URLConnection connection = (URLConnection) ProxyConfiguration.open(url);
+                connection.setConnectTimeout(20000); // JENKINS-33557 lower connection test timeout (20 sec)
```

Comment:
Ah, hadn't added the `ProxyConfiguration` value at that point, whoops - good catch.